kernel/genarch/src/multiboot/multiboot.c

```diff
@@ -167,6 +168,31 @@ void multiboot_info_parse(uint32_t signature, const multiboot_info_t *info)
 	if ((info->flags & MULTIBOOT_INFO_FLAGS_MMAP) != 0)
 		multiboot_memmap(info->mmap_length,
 		    (multiboot_memmap_t *) MULTIBOOT_PTR(info->mmap_addr));
+
+#ifdef CONFIG_FB
+
+	/* Initialize framebuffer. */
+	if ((info->flags & MULTIBOOT_INFO_FLAGS_FB) != 0) {
+		if (info->framebuffer_type != 1) {
+			/* Can't use this framebuffer. */
+			// FIXME: framebuffer_type == 2 is EGA mode, we should be able to use that.
```

Comment:
Why not do that if also CONFIG_EGA is enabled?

Reply:
Because nobody did it yet. We don't even need CONFIG_EGA as a configuration option, if the EGA framebuffer is properly configurable at runtime.

Reply:
Well we need it at least to control what support is the kernel built with.